platform/admin/core/admin-core-appservice/src/main/java/org/codice/ddf/admin/application/service/impl/ApplicationImpl.java

```diff
@@ -62,25 +62,15 @@
      *             object.
      */
     public ApplicationImpl(Repository repo) {
-        location = repo.getURI();
+        String repoName = null;
         try {
-            String[] parts = repo.getName()
-                    .split("-[0-9]");
-            if (parts.length != 0) {
-                name = parts[0];
-                version = repo.getName()
-                        .substring(name.length() + 1);
-            } else {
-                name = repo.getName();
-                version = "0.0.0";
-            }
-            features = new HashSet<>(Arrays.asList(repo.getFeatures()));
+            repoName = repo.getName();
+            features.addAll(Arrays.asList(repo.getFeatures()));
         } catch (Exception e) {
-            LOGGER.warn(
-                    "Error occured while trying to parse information for application. Application created but may have missing information.");
```

Comment:
We're throwing an exception instead of logging in every case. No one is catching the exception and it gets logged. I felt it deserved more than a warning because an exception means an application cannot be installed.